webapp/src/components/harbour/QuotaTokenBalance.test.tsx

```diff
@@ -0,0 +1,67 @@
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/react";
+import type { JsonRpcApiProvider } from "ethers";
+import type { PropsWithChildren } from "react";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+
+const signerAddress = "0xSigner";
+const quotaManagerAddress = "0xQuotaManager";
+const harbourProvider = {} as unknown as JsonRpcApiProvider;
+const className = "test-class";
+
+describe("QuotaTokenBalance", () => {
+	beforeEach(() => {
+		vi.resetModules();
+	});
+
+	it("renders token info, locked tokens, and top up form", async () => {
+		vi.doMock("@/hooks/useBrowserProvider", () => ({
+			useBrowserProvider: () => ({}),
+		}));
+		vi.doMock("@/hooks/useQuotaStats", () => ({
+			useQuotaTokenStats: () => ({
+				quotaTokenStats: {
+					tokenInfo: {
+						name: "TestToken",
+						symbol: "TTK",
+						address: "0x0000000000000000000000000000000000000001",
+						decimals: 18,
+						balance: "1000000000000000000",
+					},
+					lockedTokens: "500000000000000000",
+				},
+				isLoading: false,
+			}),
+		}));
+		vi.doMock("../Groups", () => ({
```

Comment:
Is it good practice to mock used components?

Reply:
Also should we use the `@` annotation to define what should be mocked (as you did above)